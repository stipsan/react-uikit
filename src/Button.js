import cx from 'classnames'
import { Component, PropTypes } from 'react'

export default class Button extends Component {

  static propTypes = {
    active: PropTypes.bool,
    className: PropTypes.string,
    danger: PropTypes.bool,
    large: PropTypes.bool,
    link: PropTypes.bool,
    primary: PropTypes.bool,
    secondary: PropTypes.bool,
    small: PropTypes.bool,
    text: PropTypes.bool,
  }

  static defaultProps = {
    type: 'button',
    active: false,
    primary: false,
    danger: false,
    link: false,
    small: false,
    large: false,
    className: '',
    secondary: false,
    text: false,
  }

  render() {
    const {
      active,
      primary,
      danger,
      link,
      small,
      large,
      secondary,
      text,
      className: customClassName,
      ...other
    } = this.props
    const className = cx('uk-button', customClassName, {
      'uk-button-default': !primary && !secondary && !danger && !text && !link,
      'uk-button-primary': primary,
      'uk-button-secondary': secondary,
      'uk-button-danger': danger,
      'uk-button-text': text,
      'uk-button-link': link,
      'uk-button-small': small,
      'uk-button-large': large,
      'uk-active': active,
    })

    return (
      <button {...other} className={className} />
    )
  }
}
