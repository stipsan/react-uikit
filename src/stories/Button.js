import { storiesOf, action } from '@kadira/storybook'
import { Button } from 'uikit-react'

// This is to work around: https://github.com/kadirahq/react-storybook-addon-info/issues/26#issuecomment-229029177
Button.displayName = 'Button'

storiesOf('Button', module)
  .addWithInfo('Basic Usage', '', () => (
    <div className="uk-margin-bottom">
      <Button onClick={action('onClick Default')}>Default</Button>
      &nbsp;
      <Button disabled>Default Disabled</Button>
    </div>
  ), { header: false, inline: true, propTables: [Button] })

  .addWithInfo('Style modifiers', '', () => (
    <div className="uk-margin-bottom">
      <Button onClick={action('onClick Default')}>Default</Button>
      &nbsp;
      <Button primary onClick={action('onClick Primary')}>Primary</Button>
      &nbsp;
      <Button secondary onClick={action('onClick Primary')}>Secondary</Button>
      &nbsp;
      <Button danger onClick={action('onClick Danger')}>Danger</Button>
      &nbsp;
      <Button text onClick={action('onClick Danger')}>Text</Button>
      &nbsp;
      <Button link onClick={action('onClick Link')}>Link</Button>
      &nbsp;
    </div>
  ), { header: false, inline: true, propTables: [Button] })

  .addWithInfo('Size modifiers', '', () => (
    <div className="uk-margin-bottom">
      <Button small onClick={action('onClick Success')}>SMALL BUTTON</Button>
      &nbsp;
      <Button primary small onClick={action('onClick Success')}>SMALL BUTTON</Button>
      &nbsp;
      <Button secondary small onClick={action('onClick Success')}>SMALL BUTTON</Button>
      &nbsp;
      <br /><br />
      <Button large onClick={action('onClick Submit')}>LARGE BUTTON</Button>
      &nbsp;
      <Button large primary onClick={action('onClick Submit')}>LARGE BUTTON</Button>
      &nbsp;
      <Button large secondary onClick={action('onClick Submit')}>LARGE BUTTON</Button>
      &nbsp;
    </div>
  ), { header: false, inline: true, propTables: [Button] })
