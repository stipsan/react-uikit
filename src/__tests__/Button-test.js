import renderer from 'react-test-renderer'
import Button from '../Button'

test('Button renders correctly', () => {
  expect(renderer.create(
    <Button>Submit</Button>
  ).toJSON()).toMatchSnapshot()

  expect(renderer.create(
    <Button active>Submit</Button>
  ).toJSON()).toMatchSnapshot()

  expect(renderer.create(
    <Button type="submit">Save</Button>
  ).toJSON()).toMatchSnapshot()

  expect(renderer.create(
    <Button primary>Submit</Button>
  ).toJSON()).toMatchSnapshot()

  expect(renderer.create(
    <Button success>Submit</Button>
  ).toJSON()).toMatchSnapshot()

  expect(renderer.create(
    <Button danger>Submit</Button>
  ).toJSON()).toMatchSnapshot()

  expect(renderer.create(
    <Button link>Submit</Button>
  ).toJSON()).toMatchSnapshot()

  expect(renderer.create(
    <Button small>Submit</Button>
  ).toJSON()).toMatchSnapshot()

  expect(renderer.create(
    <Button large>Submit</Button>
  ).toJSON()).toMatchSnapshot()

  expect(renderer.create(
    <Button disabled large className="save-button">Save</Button>
  ).toJSON()).toMatchSnapshot()
})
