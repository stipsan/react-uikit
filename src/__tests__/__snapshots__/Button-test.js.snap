// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="uk-button uk-button-default"
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly 2`] = `
<button
  className="uk-button uk-button-default uk-active"
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly 3`] = `
<button
  className="uk-button uk-button-default"
  type="submit"
>
  Save
</button>
`;

exports[`Button renders correctly 4`] = `
<button
  className="uk-button uk-button-primary"
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly 5`] = `
<button
  className="uk-button uk-button-default"
  success={true}
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly 6`] = `
<button
  className="uk-button uk-button-danger"
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly 7`] = `
<button
  className="uk-button uk-button-link"
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly 8`] = `
<button
  className="uk-button uk-button-default"
  mini={true}
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly 9`] = `
<button
  className="uk-button uk-button-default uk-button-small"
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly 10`] = `
<button
  className="uk-button uk-button-default uk-button-large"
  type="button"
>
  Submit
</button>
`;

exports[`Button renders correctly 11`] = `
<button
  className="uk-button save-button uk-button-default uk-button-large"
  disabled={true}
  type="button"
>
  Save
</button>
`;
